tests/test_equine_gp.py

```diff
@@ -154,3 +120,47 @@ def test_equine_gp_save_load_with_vis(random_dataset) -> None:
 
     if os.path.exists(tmp_filename):
         os.remove(tmp_filename)  # Cleanup
+
+
+@given(random_dataset=random_dataset())
+@settings(deadline=None, max_examples=1)
+def test_equine_gp_save_load_with_feature_and_label_names(random_dataset) -> None:
+    dataset, num_classes, X, embedding_model = use_basic_embedding_model(random_dataset)
+
+    # without feature and label names
+    model = eq.EquineGP(embedding_model, num_classes, num_classes)
+    loss_fn = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(
+        model.parameters(),
+        lr=0.001,
+        momentum=0.9,
+        weight_decay=0.0001,
+    )
+    model.train_model(dataset, loss_fn, optimizer, 10)
```

Comment:
Change the number epochs to like 2 or something small since we don't care about training here, and label it for clarity